src/main/java/com/minecolonies/blocks/BlockHutField.java

```diff
@@ -0,0 +1,239 @@
+    }
+
+    @Override
+    public boolean isFullCube()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isPassable(IBlockAccess worldIn, BlockPos pos)
+    {
+        return false;
+    }
+
+    @Override
+    public void onBlockPlacedBy(World worldIn, BlockPos pos, IBlockState state, EntityLivingBase placer, ItemStack stack)
+    {
+        //Only work on server side.
+        if(worldIn.isRemote)
+        {
+            return;
+        }
+
+        if(placer instanceof EntityPlayer)
+        {
+            Colony colony = ColonyManager.getColony(worldIn, pos);
+
+            if (colony != null)
+            {
+                InventoryField inventoryField = new InventoryField(LanguageHandler.getString("com.minecolonies.gui.inventory.scarecrow"), true);
+
+                ((ScarecrowTileEntity)worldIn.getTileEntity(pos)).setInventoryField(inventoryField);
+                colony.addNewField(inventoryField, ((EntityPlayer) placer).inventory, pos, worldIn);
+            }
+        }
+    }
+
+    @Override
+    public boolean isOpaqueCube()
+    {
+        return false;
+    }
+
+    @Override
+    public int getRenderType()
+    {
+        return -1;
+    }
+
+    @Override
+    public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumFacing side, float hitX, float hitY, float hitZ)
+    {
+        //If the world is server, open the inventory of the field.
+        if(!worldIn.isRemote)
+        {
+            Colony colony = ColonyManager.getColony(worldIn, pos);
+            if(colony != null)
+            {
+                playerIn.openGui(MineColonies.instance, 0, worldIn, pos.getX(), pos.getY(), pos.getZ());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void onBlockDestroyedByPlayer(final World worldIn, final BlockPos pos, final IBlockState state)
+    {
+        super.onBlockDestroyedByPlayer(worldIn, pos, state);
+        //Only work on server side.
+        if(worldIn.isRemote)
+        {
+            return;
+        }
+
+
+        Colony colony = ColonyManager.getColony(worldIn, pos);
+
+        if (colony != null)
+        {
+            colony.removeField(pos);
+        }
+    }
+
+    @Override
+    public boolean hasTileEntity(final IBlockState state)
+    {
+        return true;
+    }
+
+    // =======================================================================
+    // ======================= Rendering & IBlockState =======================
+    // =======================================================================
+    @Override
+    public IBlockState onBlockPlaced(World worldIn, BlockPos pos, EnumFacing facing, float hitX, float hitY, float hitZ, int meta, EntityLivingBase placer)
+    {
+        EnumFacing enumFacing = (placer == null) ? EnumFacing.NORTH : EnumFacing.fromAngle(placer.rotationYaw);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'enumFacing' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)